cumulus/parachains/integration-tests/emulated/common/src/macros.rs

```diff
@@ -120,3 +120,107 @@ macro_rules! test_parachain_is_trusted_teleporter {
 		}
 	};
 }
+
+#[macro_export]
+macro_rules! include_penpal_create_foreign_asset_on_asset_hub {
+	( $penpal:ident, $asset_hub:ident, $relay_ed:expr, $weight_to_fee:expr) => {
+		$crate::impls::paste::paste! {
+			pub fn penpal_create_foreign_asset_on_asset_hub(
+				asset_id_on_penpal: u32,
+				foreign_asset_at_asset_hub: MultiLocation,
+				ah_as_seen_by_penpal: MultiLocation,
+				is_sufficient: bool,
+				asset_owner: AccountId,
+				prefund_amount: u128,
+			) {
+				use frame_support::weights::WeightToFee;
+				let ah_check_account = $asset_hub::execute_with(|| {
+					<$asset_hub as [<$asset_hub Pallet>]>::PolkadotXcm::check_account()
+				});
+				let penpal_check_account =
+					$penpal::execute_with(|| <$penpal as [<$penpal Pallet>]>::PolkadotXcm::check_account());
+				let penpal_as_seen_by_ah = $asset_hub::sibling_location_of($penpal::para_id());
+
+				// prefund SA of Penpal on AssetHub with enough native tokens to pay for creating
+				// new foreign asset, also prefund CheckingAccount with ED, because teleported asset
+				// itself might not be sufficient and CheckingAccount cannot be created otherwise
+				let sov_penpal_on_ah = $asset_hub::sovereign_account_id_of(penpal_as_seen_by_ah);
+				$asset_hub::fund_accounts(vec![
+					(sov_penpal_on_ah.clone().into(), $relay_ed * 100_000_000_000),
+					(ah_check_account.clone().into(), $relay_ed * 1000),
+				]);
+
+				// prefund SA of AssetHub on Penpal with native asset
+				let sov_ah_on_penpal = $penpal::sovereign_account_id_of(ah_as_seen_by_penpal);
+				$penpal::fund_accounts(vec![
+					(sov_ah_on_penpal.into(), $relay_ed * 1_000_000_000),
+					(penpal_check_account.clone().into(), $relay_ed * 1000),
+				]);
+
+				// Force create asset on $penpal and prefund [<$penpal Sender>]
+				$penpal::force_create_and_mint_asset(
+					asset_id_on_penpal,
+					ASSET_MIN_BALANCE,
+					is_sufficient,
+					asset_owner,
+					None,
+					prefund_amount,
+				);
+
+				let require_weight_at_most = Weight::from_parts(1_100_000_000_000, 30_000);
+				let origin_kind = OriginKind::Xcm;
+				let sov_penpal_on_ah_as_location = MultiLocation {
+					parents: 0,
+					interior: X1(
+						AccountId32Junction { network: None, id: sov_penpal_on_ah.clone().into() }
+					),
+				};
```

Comment:
You don't need to create this location, just use the location for the chain and the sovereign account will be calculated correctly to deposit it there.